static/script/devices/device.js

```diff
@@ -635,14 +635,58 @@ define(
              */
             loadAuthenticatedURL: function(/*url, opts*/) {
             },
+
+            /**
+             * Returns a new XMLHttpRequest. Overridden in unit tests.
+             * @returns {XMLHttpRequest} a new instance
+             * @private
+             */
+            _newXMLHttpRequest: function() {
+                return new XMLHttpRequest();
+            },
+
             /**
              * Loads a resource from a URL.
              * @param {String} url The URL to load.
              * @param {Object} opts Object containing onLoad and onError callback functions.
-             * @returns The request object used to load the resource.
-             */
-            loadURL: function(/*url, opts*/) {
+             * @returns {XMLHttpRequest} The request object used to load the resource.
+             */
+            loadURL: function (url, opts) {
+                var xhr = this._newXMLHttpRequest();
+                xhr.onreadystatechange = function () {
+                    if (xhr.readyState === 4) {
```

Comment:
Feel like we could do with a comment or unit test to make sure someone doesn't come along and decide that using 'this' instead of 'xhr' would be better and break xhrthiswindow devices.